### DISCUSSION TITLE
Potential for a GDNative port?

### DISCUSSION THREAD

**Discussion by willnationsdev** [GodotECS/godex#173] [General]
Hey, just curious what potential there is for porting this functionality to a GDNative plugin? From the documentation I've read and twitter posts I've seen, it appears like most of the functionality is written with your own custom data structures and C++ logic. So, aside from the patch that is needed for the engine source code and the installation of the module itself, are there any other dependencies that the source code has on `/core`-only features that wouldn't be possible through the scripting API?

I ask because, in the future, I'd like to be able to use Godex for open-source addons that supplement game development where other people would also make use of those addons and help maintain them. But that is only viable if it is compatible with the vanilla Godot deployment. That of course means patches and modules can't be a dependency of the addons in question.

Is this planned eventually or is it simply too much of a development burden, etc.? I can understand if not.

**Comment by AndreaCatania** [comment]
I've converted this to a discussion, so we can keep better track of this.

Use GDNative would mean add a lot of overhead and boilerplate to it, that I don't plan to add in Godex. The idea is to keep this the more possible integrated in Godot, and GDNative would make this just too difficult or impossible. Right now I don't need much else than the minimal patch, because I'm doing it in module.

However, if someone else is willing to create an external GDNative plugin that imports Godex and allow to use it, or part of it, that's fine for me.

**Comment by willnationsdev** [comment]
Looking over the new [GDNativeExtensions API](https://github.com/godotengine/godot/pull/49744/files) submitted by reduz, with its capacity for direct ClassDB registration even at the granularity of `core`, `server`, `scene`, and `editor` levels, I wonder if a deeper level of integration may be possible in the future with GDNative. Just thought I'd mention it, for anyone else who wanders into this discussion.

**Reply by AndreaCatania** [reply]
Yee, It seems a bit more encouraging to use. It still bloating the code with native conversions and makes the compilation complex, but maybe with this changes would be easier to supports both versions: Static compilation and Dynamic compilation; so one can use the version it prefer.
